subprojects/core/src/main/java/org/gradle/internal/buildevents/TaskExecutionLogger.java

```diff
@@ -23,13 +23,15 @@
 import org.gradle.internal.logging.progress.ProgressLogger;
 import org.gradle.internal.logging.progress.ProgressLoggerFactory;
 import org.gradle.internal.progress.LoggerProvider;
+import org.gradle.internal.scan.UsedByScanPlugin;
 
 import java.util.HashMap;
 import java.util.Map;
 
 /**
  * A listener which logs the execution of tasks.
  */
+@UsedByScanPlugin
```

Comment:
It's used for not serializing ProgressStartEvent events corresponding to the task execution logger. We're only checking the package and name of the class, but a refactor of that would break our code. I'll add a comment into the annotation.